strapi/strapi
# Fix session middleware not migrated with v4

### What does it do?

* Fix unable to access the /api/connect/* routes due to the session middleware not being available
* Migrate the session middleware to v4

### Why is it needed?

The auth provider system rely on the session middleware to create and maintain user sessions.
In v4.0.0, the auth provider is broken as reported in #11337.

### How to test it?

* Start a strapi server
* Enable any login provider (I initially had the issue with auth0, #11337 mention other login provider being broken too).
* Access <stripe-api>/api/connect/<login-provider> (e.g. http://localhost:1337/api/connect/google)

If the issue persists, then you will be met by 400 error and the message "Grant: mount session middleware first".
![](https://user-images.githubusercontent.com/11737572/140897296-e7b06fd8-eea9-4f63-bda2-49b6ce016b27.png)

Following this PR, the login provider should behave correctly and create an end-user session.

### Related issue(s)/PR(s)

Related to #11337 by fixing the task "Error: Grant: mount session middleware first". Other tasks are left untouched.
Related to PR https://github.com/strapi/documentation/pull/550.
